### PR TITLE
fix: code generating enum values containing !, =, ~ are stripped

### LINKED ISSUE #229: non alpha-numeric enum values cannot be imported
For example: https://github.com/prometheus-operator/kube-prometheus/blob/373e2b415bd0ddbcb049dea54db80c2a78b76945/manifests/setup/0alertmanagerConfigCustomResourceDefinition.yaml#L67-L71

This results in:

```ts
/**
 * Match operation available with AlertManager >= v0.22.0 and takes precedence over Regex (deprecated) if non-empty.
 *
 * @schema AlertmanagerConfigSpecInhibitRulesSourceMatchMatchType
 */
export enum AlertmanagerConfigSpecInhibitRulesSourceMatchMatchType {
  /** __stripped_by_cdk8s__ */
  STRIPPED_BY_CDK8S = '__stripped_by_cdk8s__',
  /** __stripped_by_cdk8s__ */
  STRIPPED_BY_CDK8S = '__stripped_by_cdk8s__',
  /** __stripped_by_cdk8s__ */
  STRIPPED_BY_CDK8S = '__stripped_by_cdk8s__',
  /** __stripped_by_cdk8s__ */
  STRIPPED_BY_CDK8S = '__stripped_by_cdk8s__',
}
```

The reason is that these characters are not allowed as enum values. 
However, even if we do allow them, it still won't work:

```ts
/**
 * Match operation available with AlertManager >= v0.22.0 and takes precedence over Regex (deprecated) if non-empty.
 *
 * @schema AlertmanagerConfigSpecInhibitRulesSourceMatchMatchType
 */
export enum AlertmanagerConfigSpecInhibitRulesSourceMatchMatchType {
  /** != */
  VALUE_ = '!=',
  /** = */
  VALUE_ = '=',
  /** =~ */
  VALUE_ = '=~',
  /** !~ */
  VALUE_ = '!~',
}
```

This happens in [json2jsii](https://github.com/cdklabs/json2jsii/blob/main/src/type-generator.ts#L464), because we cannot deduce a member name. 

So solving this issue would require:

1. Make `json2jsii` support these enum values and append the value index to the member name in such cases.
2. Allow these characters in the cdk8s cli.

### PR DESCRIPTION
**Pull request by vinayak-kukreja**
This has been fixed in json2jsii. Allowlisting characters `!, =, ~`. 

Fixes https://github.com/cdk8s-team/cdk8s-cli/issues/229

### CODE REVIEW

File: test/import/fixtures/kube-prometheus.yaml
Comment (vinayak-kukreja) [inline test/import/fixtures/kube-prometheus.yaml:58]: These are now working. Unfortunately the diff is huge so it does not show up in the web ui. But, you can take this locally and verify it. 

This is what it looks now,

```
/**
 * Match operation available with AlertManager >= v0.22.0 and takes precedence over Regex (deprecated) if non-empty.
 *
 * @schema AlertmanagerConfigSpecInhibitRulesSourceMatchMatchType
 */
export enum AlertmanagerConfigSpecInhibitRulesSourceMatchMatchType {
  /** != */
  VALUE_NOT_EQUALS_TO = \\"!=\\",
  /** = */
  VALUE_ = \\"=\\",
  /** =~ */
  VALUE_EQUAL_TILDE = \\"=~\\",
  /** !~ */
  VALUE_NEGATION_TILDE = \\"!~\\",
}
```

Comment (vinayak-kukreja) [reply test/import/fixtures/kube-prometheus.yaml:58]: `  VALUE_ = \\"=\\",`

Fixing this in Json2Jsii now

Comment (paulhcsun) [reply test/import/fixtures/kube-prometheus.yaml:58]: What is the purpose of this file?

Comment (vinayak-kukreja) [reply test/import/fixtures/kube-prometheus.yaml:58]: So this is a Custom Resource Definition that is failing in the issue mentioned. We can add these to the tests and they create a snapshot that we can verify if the generated code is as expected. 

This CRD was failing code generation due to symbols that were not handled correctly by Json2Jsii. So I fixed that first and then allowlisted these in cdk8s. 

Since `=` was missed, the code generation just gave us `VALUE_` which is incorrect. So I will update the json2jsii version with the fix you just approved and that should fix this. Until then we cannot merge this.

**Review by paulhcsun — APPROVED** [review]

File: test/import/fixtures/kube-prometheus.yaml
Comment (paulhcsun) [reply test/import/fixtures/kube-prometheus.yaml:58]: lgtm, thanks for explaining. Added a do-not-merge tag, feel free to remove and merge once json2jsii version is updated

Comment (vinayak-kukreja) [reply test/import/fixtures/kube-prometheus.yaml:58]: Great, thank you.

After the update it is,
```
/**
 * Match operation available with AlertManager >= v0.22.0 and takes precedence over Regex (deprecated) if non-empty.
 *
 * @schema AlertmanagerConfigSpecInhibitRulesSourceMatchMatchType
 */
export enum AlertmanagerConfigSpecInhibitRulesSourceMatchMatchType {
  /** != */
  VALUE_NOT_EQUALS_TO = \\"!=\\",
  /** = */
  VALUE_EQUALS = \\"=\\",
  /** =~ */
  VALUE_EQUAL_TILDE = \\"=~\\",
  /** !~ */
  VALUE_NEGATION_TILDE = \\"!~\\",
}
```

**Pull request merged**
